### DISCUSSION TITLE
HOWTO using 4pin PC fan

### DISCUSSION THREAD

**Discussion by TrDA-hab** [arendst/Tasmota#13090] [Show and tell]
# 0. About using 4pin PC fan.

- Using [PWM](https://tasmota.github.io/docs/Commands/#pwm).
- Using [MultiPwm](https://tasmota.github.io/docs/Commands/#setoption68).
- Using [DC18B20](https://tasmota.github.io/docs/DS18x20/).  
- Using [Thermostat](https://tasmota.github.io/docs/Thermostat/).
- About sizes [PC fan](https://digitalworld839.com/computer-case-fan-sizes/).  
- About [PC fan](https://en.wikipedia.org/wiki/Computer_fan) (wikipedia).  
- About [Fan speed regulators](https://www.maximintegrated.com/en/design/technical-documents/tutorials/1/1784.html).
- [PC Fan datasheet](https://noctua.at/pub/media/wysiwyg/Noctua_PWM_specifications_white_paper.pdf).
- [# 9238](https://github.com/arendst/Tasmota/issues/9238#issuecomment-749299589) (Home Assistant - no tested).
- [# 11257](https://github.com/arendst/Tasmota/discussions/11257#discussioncomment-450144) (Tachometer - no tested). 
- [# 3053](https://github.com/arendst/Tasmota/issues/3053#issuecomment-400180836 ) (Tachometer - no tested).   

### Practical use of the PC fan.     
- An [example](https://www.thingiverse.com/thing:4163250) of the practical use of the 4pin PC fan for home ventilation.      
- An [example](https://www.instructables.com/Fan-Assembly-for-Fireplaces-and-Heat-Radiators/) of the fan assembly for Fireplaces and Heat Radiators.      

### 4pin PC fan Wire Diagrams.     
*!!! cable coloring varies from fan to fan*   

Pin|Name|Color-1|Color-2|Color-3|Color-4|  
:-:|:-:|:-:|:-:|:-:|:-:|  
|1|GND|![](https://img.shields.io/static/v1?label=&message=BLACK&color=black)|![](https://img.shields.io/static/v1?label=&message=BLACK&color=black)|![](https://img.shields.io/static/v1?label=&message=GREY&color=lightgrey)|![](https://img.shields.io/static/v1?label=&message=BLACK&color=black)|   
|2|+12VDC|![](https://img.shields.io/static/v1?label=&message=RED&color=red)|![](https://img.shields.io/static/v1?label=&message=BLACK&color=black)|![](https://img.shields.io/static/v1?label=&message=GREY&color=lightgrey)|![](https://img.shields.io/static/v1?label=&message=YELLOW&color=yellow)|   
|3|TACH|![](https://img.shields.io/static/v1?label=&message=YELLOW&color=yellow)|![](https://img.shields.io/static/v1?label=&message=BLACK&color=black)|![](https://img.shields.io/static/v1?label=&message=GREY&color=lightgrey)|![](https://img.shields.io/static/v1?label=&message=GREEN&color=greem)   
|4|PWM|![](https://img.shields.io/static/v1?label=&message=BLUE&color=blue)|![](https://img.shields.io/static/v1?label=&message=BLACK&color=black)|![](https://img.shields.io/static/v1?label=&message=GREY&color=lightgrey)|![](https://img.shields.io/static/v1?label=&message=BLUE&color=blue)|   

![](https://raw.githubusercontent.com/TrDA-hab/Projects/master/PC%20fan/500-4.jpg)  

# 1. Minimum configuration:  

![](https://raw.githubusercontent.com/TrDA-hab/Projects/master/PC%20fan/501.jpg)  

# 2. Midle configuration:  

![](https://raw.githubusercontent.com/TrDA-hab/Projects/master/PC%20fan/502.jpg)  

# 3. Maximum configuration:  

### Note:
Now (09/07/2021) Tasmota has no support for the TAСH (tachometer) sensor. Maybe support for the TAСH will be added at a later date.

![](https://raw.githubusercontent.com/TrDA-hab/Projects/master/PC%20fan/503.jpg)  
![](https://raw.githubusercontent.com/TrDA-hab/Projects/master/PC%20fan/504.jpg)  

**Best regards,  
TrDA.**

**Comment by barbudor** [comment]
Hi @TrDA-hab 
What kind of solid-state-relays are you using ?
Most of those I know are for an AC load

**Reply by TrDA-hab** [reply]
I bought them a long time ago, but Google knows where to find them now.
https://ae01.alicdn.com/kf/ULB8BN.dpf2JXKJkSanrq6y3lVXai.jpg
https://ae01.alicdn.com/kf/ULB8Y9bxImnEXKJk43Ubq6zLppXaM.jpg

https://aliexpress.com/item/4001106299993.html
https://aliexpress.com/item/4000074441199.html
https://www.ebay.fr/itm/173033320962

**Reply by TrDA-hab** [reply]
This module is called "DC-DC PCB SSR":
https://aliexpress.com/item/4001095559640.html
https://aliexpress.com/item/32918232923.html
https://aliexpress.com/item/32621378346.html
https://aliexpress.com/item/32701663642.html
https://aliexpress.com/item/1005001345092066.html
https://www.ebay.com/itm/401288032144
https://www.ebay.com/itm/191211053635

**Reply by barbudor** [reply]
Spasibo 👍 
Now, I found this shop with a large selection: https://www.aliexpress.com/store/1240331/search?origin=y&SearchText=SSR

**Reply by TrDA-hab** [reply]
https://aliexpress.com/item/32671829582.html

**Reply by barbudor** [reply]
30A should be good for a PC fan 😂

**Reply by joba-1** [reply]
![image](https://user-images.githubusercontent.com/32450554/132990056-a93fbad9-d34f-4250-92dc-ac7f1a36a027.png)
maybe not…

**Reply by TrDA-hab** [reply]
> 30A should be good for a PC fan 😂


why not  )))

![](https://i.redd.it/8ytoya89vo221.jpg)  
![](https://www.deatonengineering.com/images/photo_fan_test1.jpg)  
![](https://content.instructables.com/ORIG/FZ1/GP50/J8YRMIC4/FZ1GP50J8YRMIC4.jpg)  
![](https://content.instructables.com/ORIG/FG2/1OHV/I03OBHTW/FG21OHVI03OBHTW.jpg)  
![](https://www.brilliantdiy.com/wp-content/uploads/2016/11/fan_system.png)
![](https://content.instructables.com/ORIG/FJU/SNXI/I3H57B7C/FJUSNXII3H57B7C.jpg)

**Comment by DREEMus** [comment]
Guys, no one seen that DS18B20 are connected to 12V directly in midle configuration?

**Reply by TrDA-hab** [reply]
No. It must be connected to 3.3V.

**Reply by DREEMus** [reply]
Look once again on midle configuration, DSs are separated from 3.3V and are connected to 12V directly with fans.

**Comment by drcyberg** [comment]
Hi Masters!
I have a question about the 504.jpg circuit diagram:
Doesn't the TACH pin require a 4K7 3.3V pull-up resistor?
Isn't the signal only interpretable if there is an external pull-up?
Thank you!
Regards!

**Reply by sfromis** [reply]
By default, Tasmota will enable the ESP weak internal pull-up on most types of input pins, unless you explicitly configure to not do this. Hence no *external* pull-up needed.
